funsor/terms.py

```diff
@@ -59,7 +46,7 @@ def alpha_convert(expr):
     new_values = []
     for v in expr._ast_values:
         v = substitute(v, alpha_subs)
-        if isinstance(v, str):
+        if isinstance(v, str) and v not in expr.fresh:
```

Comment:
Can you explain when the new condition is triggered? Thanks for the help.

Reply:
Since `Independent` has a fresh variable with the same name as a bound one, this check prevents `alpha_convert` from renaming `Independent`'s `reals_var` attribute.

Reply:
OK, let's just keep this edge case pattern in mind.